src/redis.c

```diff
@@ -140,11 +142,18 @@ static redis_query_t *redis_config_query(oconfig_item_t *ci) /* {{{ */
   for (int i = 0; i < ci->children_num; i++) {
     oconfig_item_t *option = ci->children + i;
 
+    &rq->database = 0;
+
     if (strcasecmp("Type", option->key) == 0) {
       status = cf_util_get_string_buffer(option, rq->type, sizeof(rq->type));
     } else if (strcasecmp("Instance", option->key) == 0) {
       status =
           cf_util_get_string_buffer(option, rq->instance, sizeof(rq->instance));
+    } else if (strcasecmp("Database", option->key) == 0) {
+      status = cf_util_get_int(option, &rq->database);
+      if (option[0].value.number < 0) {
```

Comment:
1) Use `rq->database` here.
2) Add `WARNING('Message')` explaining what is happening.

Reply:
you right :-(
but i change in redis_handle_query already.

if it's problem — i can re-edit again :)